### PR TITLE
Feat: fee refunder module documentation

### PR DESCRIPTION
**Pull request by pr0n00gler**

### CODE REVIEW

File: docs/neutron/feerefunder/overview.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
+# Overview
+
+## Abstract
+
+This document specifies the FeeRefunder module for the Neutron network.
+
+The FeeRefunder module implements a mechanism and contains methods to make refunder IBC relayers for their submission of IBC `Ack` and `Timeout` packets.
```
Comment (oopcode) [inline docs/neutron/feerefunder/overview.md:7]: to make refunds to IBC relayers?

File: docs/neutron/feerefunder/overview.md [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+## Concepts
+
+Due to the fact that contracts are allowed to make calls to the IBC as well as process all received data, a problem arieses in which a malicious contract can make a call to the IBC and, during the response of the `Ack` or `Timeout`, make another IBC call and so on forever. Which will lead to drainage of relayer's funds and spamming of the network.
+
+In order to avoid such problem, the code in the module forces the contract to pay for all Acknoledgements and Timeout messages it processes.
+
+The mechanism behind the module is inspired by [ICS-29](https://github.com/cosmos/ibc/tree/main/spec/app/ics-029-fee-payment). ICS-29 itself requires both chains (target and source chains) to support the specifications which is not quite good for Neutron, since we want to connect as many chains as possible, and that's hard with ICS-29, since not so many chains are upgrade to the latest [ibc-go](https://github.com/cosmos/ibc-go) version.
```
Comment (oopcode) [inline docs/neutron/feerefunder/overview.md:15]: use

File: docs/neutron/feerefunder/overview.md [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+That is why we decided to implement our own IBC Fees module, which allows to refund relayers for their `Ack/Timeout` submission, but with the same interface for the smart-contract developers. So when Neutron will support native ICS-29, there will no need for smart-contract developers to change their contracts, upgrade of the fees module will be very smooth.
```
Comment (oopcode) [inline docs/neutron/feerefunder/overview.md:17]: upgrading to the "official" fees module will be smooth.

File: docs/neutron/feerefunder/overview.md [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+* `timeout_fee` - amount of coins to refund relayer for submitting timeout message for a particular IBC packet (i.e. `500untrn`);
+* `recv_fee` - currently is used for compatibility with ICS-29 interface only and **must be set to zero** (i.e. `0untrn`), because Neutron's fee module can't refund relayers for submission of `Recv` IBC packets due to compatibility with target chains.
+
+> **Note:** the fees can be specified only in native Cosmos-SDK coins. CW-20 coins are not supported!
+
+When a smart-contract issues `Transfer` or `SubmitTx` message, the fee Module deduct the whole specified fee amount (`ack_fee + timeout_fee + recv_fee`) and locks that amount in the module's escrow address. When a relayer submits `Ack` message for a particular packet, the module sends the specified amount of `ack_fee` to the relayer from the escrow address and return the specified `timeout_fee` to the contract which issued the original `Transfer` or `SubmitTx` message. In case when relayer submits `Timeout` message, things go the other way around: the relayer is refunded with `timeout_fee` and the contract gets `ack_fee` back.
```
Comment (oopcode) [inline docs/neutron/feerefunder/overview.md:27]: did i miss the part about the minimal fee parameter? can't see it anywhere

File: docs/neutron/feerefunder/overview.md (hunk 1 above)
Comment (pr0n00gler) [reply docs/neutron/feerefunder/overview.md]: Done

File: docs/neutron/feerefunder/overview.md (hunk 2 above)
Comment (pr0n00gler) [reply docs/neutron/feerefunder/overview.md]: Done

File: docs/neutron/feerefunder/overview.md (hunk 3 above)
Comment (pr0n00gler) [reply docs/neutron/feerefunder/overview.md]: Done

File: docs/neutron/feerefunder/overview.md (hunk 4 above)
Comment (pr0n00gler) [reply docs/neutron/feerefunder/overview.md]: Added min fee info parameter

**Pull request merged**